src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -670,18 +671,29 @@ BinderCheckTypeResult ReportInvalidOperation()
             }
         }
 
-        private static BinderCheckTypeResult CheckComparisonArgTypesCore(IErrorContainer errorContainer, TexlNode left, TexlNode right, DType typeLeft, DType typeRight)
+        private static BinderCheckTypeResult CheckComparisonArgTypesCore(IErrorContainer errorContainer, TexlNode binaryOpNode, TexlNode left, TexlNode right, DType typeLeft, DType typeRight)
```

Comment:
>TexlNode [](http://example.com/codeflow?start=105&length=8)

Why not this be BinaryOpNode? Left and right can be retrieved from this node.

Reply:
In other places in the file we use leftNode.Parent, so I'll do that.